pyk/src/tests/integration/proof/test_imp.py

```diff
@@ -1066,6 +1066,52 @@ def test_failure_info(
 
         assert actual_path_conds == expected_path_conds
 
+    def test_proof_no_progress_on_reload(
+        self,
+        kprove: KProve,
+        kcfg_explore: KCFGExplore,
+        proof_dir: Path,
+    ) -> None:
+
+        spec_file = K_FILES / 'imp-simple-spec.k'
+        spec_module = 'IMP-SIMPLE-SPEC'
+        claim_id = 'fail-early'
+        expected_pending = 1
+        expected_failing = 1
+
+        claim = single(
+            kprove.get_claims(Path(spec_file), spec_module_name=spec_module, claim_labels=[f'{spec_module}.{claim_id}'])
+        )
+
+        proof = APRProof.from_claim(kprove.definition, claim, logs={}, proof_dir=proof_dir)
+        proof_id = proof.id
+        kcfg_explore.simplify(proof.kcfg, {})
+        prover = APRProver(kcfg_explore=kcfg_explore)
+        prover.advance_proof(proof, fail_fast=True)
+
+        failure_info = proof.failure_info
+        assert isinstance(failure_info, APRFailureInfo)
+
+        actual_pending = len(failure_info.pending_nodes)
```

Comment:
Could we have a stronger comparison, perhaps of the associated node ids? This would mean no progress, the current test only makes sure that there is one pending and one failing node, but not necessarily that it's the same ones.

Reply:
I'm comparing the whole FailureInfo object before and after the reload now, so it is now checking the exact node ids are the same, not just the number.